user_sync/post_sync/manager.py

```diff
@@ -0,0 +1,104 @@
+import logging
+import six
+from copy import deepcopy
+from .connectors import get_connector
+
+
+class PostSyncManager:
+    def __init__(self, post_sync_config, test_mode):
+        self.config = post_sync_config
+        self.logger = logging.getLogger("post-sync")
+        self.umapi_users = {}
+
+        # Assemble to connector list
+        self.connectors = [
+            get_connector(m, c, test_mode) for m, c in six.iteritems(self.config['modules'])
+        ]
+
+    def get_directory_attributes(self):
+        attributes = set()
+        for conn in self.connectors:
+            attributes |= set(conn.get_directory_attributes())
+        return attributes
+
+    def run(self, post_sync_data):
+        """
+        run each entry from the module dict from __init__
+        :return:
+        """
+        for connector in self.connectors:
+            self.logger.info("Running module " + connector.name)
+            connector.run(post_sync_data)
```

Comment:
Good opportunity for error handling to avoid hard stop if one of the post sync connectors errors out